src/base/CMakeLists.txt

```diff
@@ -167,13 +165,39 @@ target_link_libraries(qbt_base
         Qt5::Core Qt5::Network Qt5::Xml
 )
 
-if (Qt5DBus_FOUND)
-    target_link_libraries(qbt_base PRIVATE Qt5::DBus)
+if (DBUS)
+    target_link_libraries(qbt_base PUBLIC Qt5::DBus)
+endif()
+
+if (NOT COUNTRIES_RESOLUTION)
+    target_compile_definitions(qbt_base PUBLIC DISABLE_COUNTRIES_RESOLUTION)
 endif()
 
-if (APPLE)
+if (NOT GUI)
+    target_compile_definitions(qbt_base PUBLIC DISABLE_GUI)
+endif()
+
+if (NOT WEBUI)
+    target_compile_definitions(qbt_base PUBLIC DISABLE_WEBUI)
+endif()
+
+# only needed for calls to lt::generate_fingerprint and for the qbittorrent.rc file on Windows
+# the latter forces these definitions to be PUBLIC instead of PRIVATE,
+# unless we wanted to repeat the same inside src/app/CMakeLists.txt for the executable target
```

Comment:
These definitions have project-wide meaning generally. Don't understand why you moved it there.

Reply:
@glassez It's `QBT_VERSION` and `QBT_VERSION_2` that are ubiquitous. These particular ones only have meaning for "base" and "app", so I created a specific interface for that, to isolate it. In the future, I'd like to propose some versioning changes to prevent these kind of inconsistencies, but that's a matter for another time.